living_documentation_generator/utils/decorators.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from typing import Callable, Optional, Any
+from functools import wraps
+
+from living_documentation_generator.utils.github_rate_limiter import GithubRateLimiter
+
+
+def debug_log_decorator(method: Callable) -> Callable:
+    """
+    Decorator to add debug logging for a method call.
+    """
+    @wraps(method)
+    def wrapped(*args, **kwargs) -> Optional[Any]:
```

Comment:
I would rewrite type hint `Optional[Any]` into `Any | None`, so we don't have to import typing Optional at all. Same in the safe_call_decorator method.

Reply:
Optional is more readable. I want to keep it here as we are using it at more locations.